File: src/core/NodeStatus.ts
/**
 * Represents the execution status of a behavior tree node.
 *
 * - SUCCESS: The node completed successfully
 * - FAILURE: The node failed to complete its task
 * - RUNNING: The node is still executing and needs more ticks
 * - IDLE: The node has not been executed yet or has been reset
 */
export enum NodeStatus {
    SUCCESS = 'success',
    FAILURE = 'failure',
    RUNNING = 'running',
    IDLE = 'idle'
}

File: UnitySDK/Runtime/Core/NodeStatus.cs
namespace WebBehaviorTree
{
    /// <summary>
    /// Status returned by a behavior tree node after execution.
    /// </summary>
    public enum NodeStatus
    {
        /// <summary>Node has not been executed yet.</summary>
        Idle,

        /// <summary>Node is still executing and needs more time.</summary>
        Running,

        /// <summary>Node completed successfully.</summary>
        Success,

        /// <summary>Node failed to complete its task.</summary>
        Failure
    }
}
